### PR TITLE
[quant][pt2e] Refactor conv related annotation for XNNPACKQuantizer

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110308

Summary:
Since we changed IR that we are working with to pre autograd aten IR, it's easier
to use plain pattern match instead of relying on source_matcher_utils now, this
PR refactors the annotation for conv to use aten ops directly.

Also fixed reentrant test after this change.

Test Plan:
python test/test_quantization.py TestQuantizePT2E

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py [hunk 1]
```diff
@@ -268,36 +258,27 @@ def _annotate_conv_relu(
...
             torch.ops.aten.relu_.default,
         ]:
-            raise ValueError(f"{relu_node} is not an aten relu operator")
+            continue
+        relu_node = n
+        maybe_conv_node = n.args[0]
+        if not isinstance(maybe_conv_node, Node) or maybe_conv_node != "call_function" or maybe_conv_node.target not in [
+            torch.ops.aten.conv1d.default,
+            torch.ops.aten.conv2d.default,
+        ]:
```
Comment (kimishpatel) [inline torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py:273]: I think we should consider refactoring this into either
1. subgraph matcher based OR
2. refactor find_sequential_partitions or add another util there to take in a list of partitions are find those which are sequentially connected. 

The whole list of if conditions is not great.

File: test/quantization/pt2e/test_quantize_pt2e.py [hunk 2]
```diff
@@ -1861,7 +1861,7 @@ def test_reentrant(self):
         m(*example_inputs)
         m.conv_bn_relu = convert_pt2e(m.conv_bn_relu, fold_quantize=True)
 
-        quantizer = XNNPACKQuantizer().set_global(get_symmetric_quantization_config(is_per_channel=False))
+        quantizer = XNNPACKQuantizer().set_module_type(torch.nn.Linear, get_symmetric_quantization_config(is_per_channel=False))
```
Comment (kimishpatel) [inline test/quantization/pt2e/test_quantize_pt2e.py:1864]: why this change?

**Review by kimishpatel — CHANGES_REQUESTED** [review]
if possible another util like find_sequential_partition that can find fused patterns would be a better code structure.

btw did this break re-entrant test?

File: test/quantization/pt2e/test_quantize_pt2e.py (hunk 2 above)
Comment (jerryzh168) [reply test/quantization/pt2e/test_quantize_pt2e.py:1864]: this is the fix for re-entrant test

File: torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py]: so in general you'll need to get a conv_node and a relu_node, if the API you have did not give a general access to these things, it will still be hacky and making assumptions. let's say `find_connected_nodes([aten.conv2d, aten.relu], [aten.cov1d, aten.relu])`, what API will this function provide for accessing conv2d and relu node? and what if this API takes 10 ops and connected in a graph?

but this is XNNPACKQuantizer so please feel free to make any APIs, it might be better for edge team to design and own these APIs though, would you be OK to land this one first and you can follow up later to add these APIs?

Comment (kimishpatel) [reply torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py]: > what API will this function provide for accessing conv2d and relu node? and what if this API takes 10 ops and connected in a graph (instead of just sequentially)?

Thats a good point. However you just need input and outptu nodes, right? Anyhow, I do see your point. If we want to do this right, it will have to be something that can handle many cases.

I am fine landing this and we can own and update

**Review by kimishpatel — APPROVED** [review]

File: torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py]: yeah for most use cases we probably just need input/output, it's fine if you want to start with this and then expand it to more general utils. I was thinking of building this general util actually, but have to figure out the priority.
